C++/test/algorithm/backtracking/n_queens.cpp

```diff
@@ -0,0 +1,49 @@
+#include "third_party/catch.hpp"
+#include "algorithm/backtracking/n_queens.hpp"
+
+TEST_CASE("Unplaceable cases", "[backtracking][n_queens]") {
+    NQueensSolver b0(0);
+    REQUIRE_FALSE(b0.can_place_queens());
+    NQueensSolver b2(2);
+    REQUIRE_FALSE(b2.can_place_queens());
+    NQueensSolver b3(3);
+    REQUIRE_FALSE(b3.can_place_queens());
+}
+
+TEST_CASE("Placeable cases", "[backtracking][n_queens]") {
+    NQueensSolver b1(1);
```

Comment:
One thing I would consider (though I think either way is fine) is having the categories be 

- Base cases (N = 0, N = 1)
- Placable cases (N = 4, N = 9, etc.)
- Unplaceable cases (N = 2, N = 3, etc.)

This would fit with our other tests (bases cases, other cases, etc.). But again, I think what you have here works well. Just another way of looking at it.